pkg/controller/seed-controller-manager/kubernetes/resources.go

```diff
@@ -401,14 +401,20 @@ func (r *Reconciler) ensureClusterRoleBindings(ctx context.Context, c *kubermati
 
 // GetConfigMapCreators returns all ConfigMapCreators that are currently in use
 func GetConfigMapCreators(data *resources.TemplateData) []reconciling.NamedConfigMapCreatorGetter {
-	return []reconciling.NamedConfigMapCreatorGetter{
+	creators := []reconciling.NamedConfigMapCreatorGetter{
 		cloudconfig.ConfigMapCreator(data),
 		openvpn.ServerClientConfigsConfigMapCreator(data),
 		dns.ConfigMapCreator(data),
 		apiserver.AuditConfigMapCreator(),
 		apiserver.AdmissionControlCreator(data),
 		apiserver.CABundleCreator(data),
 	}
+
+	if data.Cluster().Spec.Cloud.VSphere != nil {
+		creators = append(creators, cloudconfig.ConfigmapVsphereCSICreator(data))
+	}
```

Comment:
Why we don't create a Secret instead of a ConfigMap? The original cloud-config is also a Secret if I'm not mistaken.

Reply:
The cloud-config resides in two distinct data structures: a cm in the seed cluster (created by the seed-controller-manager) and a secret in the user-cluster (created by the user-cluster-controller-manager) created from the aforementioned cm.

Reply:
here is the secret creation in the user-cluster: https://github.com/kubermatic/kubermatic/blob/6c29856f862cb6319575034bec202fb3694d235d/pkg/controller/user-cluster-controller-manager/resources/reconciler.go#L613